test/models/badges/researcher_badge_test.rb

```diff
@@ -0,0 +1,58 @@
+require "test_helper"
+
+class Badge::ResearcherBadgeTest < ActiveSupport::TestCase
+  test "attributes" do
+    badge = create :researcher_badge
+    assert_equal "Researcher", badge.name
+    assert_equal :ultimate, badge.rarity
+    assert_equal :researcher, badge.icon
+    assert_equal 'Helped develop Exercism Research', badge.description
+    assert badge.send_email_on_acquisition?
+    assert_nil badge.notification_key
+  end
+
+  test "award_to?" do
+    badge = create :researcher_badge
+
+    non_researcher_user = create :user
+    refute badge.award_to?(non_researcher_user)
+
+    non_github_user = create :user, github_username: nil
+    refute badge.award_to?(non_github_user)
```

Comment:
Added this test to ensure there is no nil exception when a user does not have a github username